src/GeneralStateTestsFiller/stReturnDataTest/tooLongReturnDataCopyFiller.yml

```diff
@@ -0,0 +1,247 @@
+tooLongReturnDataCopy:
+  _info:
+    comment: Ori Pomerantz   qbzzt1@gmail.com
+
+  env:
+    currentCoinbase: 2adc25665018aa1fe0e6bc666dac8fc2697ff9ba
+    currentDifficulty: 0x20000
+    currentNumber: 1
+    currentTimestamp: 1000
+    currentGasLimit: 0x10000000000000
+    previousHash: 5e20a0453cecd065ea59c37ac63e079ee08998b6045136a8ce6635c7912ec0b6
+    currentBaseFee: 10
+
+  pre:
+    # Return the requested amount of data
+    000000000000000000000000000000000000C0DE:
+      balance: 0
+      code: |
+        :yul {
+           // How many bytes to return
+           let byteCount := calldataload(0)
+
+           // Some junk data
+           mstore(0, 0x1F1E1D1C1B1A191817161514131211100F0E0D0C0B0A090807060504030201FF)
+
+           sstore(0, byteCount)
+
+           // Return the result
+           return(0x00, byteCount)
+        }
+      nonce: 1
+      storage: {}
+
+
+    # Revert with the requested amount of data
+    0000000000000000000000000000000000000BAD:
+      balance: 0
+      code: |
+        :yul {
+           // How many bytes to return
+           let byteCount := calldataload(0)
+
+           // Some junk data
+           mstore(0, 0x1F1E1D1C1B1A191817161514131211100F0E0D0C0B0A090807060504030201FF)
+
+           sstore(0, byteCount)
+
+           // Return the result
+           revert(0x00, byteCount)
+        }
+      nonce: 1
+      storage: {}
+
+
+    # Call 0xC0DE or 0x0BAD and then copy return data (possibly an excessive
+    # amount of it)
+    cccccccccccccccccccccccccccccccccccccccc:
+      balance: 1000000000000000000
+      code: |
+          :yul {
+             let lengthReturned := calldataload(0x04)
+             let offsetCopied   := calldataload(0x24)
+             let lengthCopied   := calldataload(0x44)
+             let contract       := calldataload(0x64)
+             mstore(0, lengthReturned)
+
+             let retVal := call(gas(), contract, 0, 
+                0, 0x20,    // input buffer with lengthReturned
```

Comment:
how come something 'returned' is an input?

Reply:
The length of the buffer to be returned is part of the calldata for this contract. However, it is necessary to send it to the contract we're calling (either 0xC0DE or 0x0BAD) so it will know what size of buffer to return to us.

Reply:
I added the explanation to the code